http-mclient/src/Network/HTTP/MClient.hs

```diff
@@ -120,34 +127,76 @@ jsonContentType = "application/json; charset=utf-8"
 jsonLegacyContentType :: BS.ByteString
 jsonLegacyContentType = "application/json"
 
+defaultTransaction :: Transaction a
+defaultTransaction = Transaction
+  { decoder     = Left . HTTPError
+  , retryPolicy = Retry.constantDelay 50000 <> Retry.limitRetries 1
+  , ..
+  }
+  where
+    shouldRetry :: HTTP.HttpExceptionContent -> Bool
+    shouldRetry = \case
+      HTTP.ConnectionFailure _ -> True
+      _                        -> False
```

Comment:
__comment:__ I know we use `\case` because it allows us to use `where`, but if we're not using `where`, maybe it would be a "simpler primitive" if we wrote this as:

```haskell
shouldRetry :: HTTP.HttpExceptionContent -> Bool
shouldRetry HTTP.ConnectionFailure _ = True
shouldRetry _                        = False
```

Reply:
@dkubb I am quite used to & fond of using `\case` for pattern matching since it's less noisy during multiple matches. However, your argument makes sense and I wouldn't mind us standardizing with a simpler primitive despite the syntax noise.